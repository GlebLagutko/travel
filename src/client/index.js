/*
import {gql, ApolloClient, InMemoryCache} from '@apollo/client';
import store from "../store";


export const client = new ApolloClient({
    uri: 'https://desolate-bastion-22611.herokuapp.com/graphql',
    cache: new InMemoryCache()
});


console.log("************")

export function clientQuery() {
    client
    .query({
        query: gql`
            query
            {
                games{user, difficulty, time}
            }
        `,
        fetchPolicy: "no-cache"
    });
}

clientQuery();
*/



