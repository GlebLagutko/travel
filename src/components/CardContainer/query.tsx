import {gql} from "@apollo/client";

export default gql`
        query Countries($language: String!){
            Countries(language: $language){
                id
                name
                video
                urlName
                capital
            }
        }`;
