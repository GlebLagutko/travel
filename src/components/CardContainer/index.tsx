import React, {useState} from "react";
import CountryCard from "../CountryCard";
import {makeStyles} from "@material-ui/core/styles";
import {useSelector} from "react-redux";
import {useQuery} from "@apollo/client";
import query from "./query";


interface ContainerProps {
    searchValue: string;
}

const useStyles = makeStyles({
    root: {
        display: "flex",
        flexWrap: "wrap",
        justifyContent: "space-around",
        maxWidth: "1200",
        columnGap: "30px",
        rowGap: "30px",
        padding: "50px 0",
    },
});


const languageState = state => state.value.language;

export default function CardContainer({searchValue}: ContainerProps) {

    const [countries, setCountries] = useState([]);
    const language = useSelector(languageState);

    useQuery(query, {
        variables: {language: language},
        onCompleted: data => {
            setCountries(data["Countries"]);
        }
    });
    //
    const cards = [];


    countries.forEach((elem) => {
        let search = searchValue.toLowerCase();
        if (elem.name.toLowerCase().match(search) || elem.capital.toLowerCase().match(search)) {
            cards.push(<CountryCard country={elem} key={elem.id}/>);
        }
    });

    const styles = useStyles();
    return (<div className={styles.root}>
        {
            cards
        }
    </div>)
}
