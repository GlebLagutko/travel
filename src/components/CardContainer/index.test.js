import {render} from '@testing-library/react';

import React from "react";
import CardContainer from ".";
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'
import {MockedProvider} from '@apollo/client/testing';
import query from "./query";

describe('CardContainer', () => {
  it('renders correctly', async () => {

    const initialState = {value: {language: "en"}};
    const mockStore = configureStore();
    let store = mockStore(initialState);

    const mocks = [
      {
        request: {
          query: query,
          variables: {
            language: 'en',
          },
        },
        result: {
          data: {
            "Countries": [
              {
                "id": 1,
                "name": "Norway",
                "video": "https://www.youtube.com/embed/Scxs7L0vhZ4",
                "urlName": "norway",
                "capital": "Oslo"
              }
            ]
          }
        },
      }
    ];

    const {container, findByText} = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Provider store={store}><CardContainer searchValue=""/></Provider>
        </MockedProvider>, );

    expect(await findByText("Norway")).toBeInTheDocument();
    expect(container).toMatchSnapshot();

  });

});