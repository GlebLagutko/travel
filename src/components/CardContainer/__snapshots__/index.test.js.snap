// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer renders correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-3"
          style="background-image: url(assets/images/thumbnail-norway.jpg);"
          title="Contemplative Reptile"
        />
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            <span>
              Norway
            </span>
          </h2>
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            <span>
              Oslo
            </span>
          </h2>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
